src/pg/pg-store.ts

```diff
@@ -0,0 +1,513 @@
+      WHERE loc.current = TRUE AND gen.genesis = TRUE
+        ${
+          args?.genesis_id?.length
+            ? this.sql`AND i.genesis_id IN ${this.sql(args.genesis_id)}`
+            : this.sql``
+        }
+        ${
+          args?.genesis_block_height
+            ? this.sql`AND gen.block_height = ${args.genesis_block_height}`
+            : this.sql``
+        }
+        ${
+          args?.genesis_block_hash
+            ? this.sql`AND gen.block_hash = ${args.genesis_block_hash}`
+            : this.sql``
+        }
+        ${
+          args?.from_genesis_block_height
+            ? this.sql`AND gen.block_height >= ${args.from_genesis_block_height}`
+            : this.sql``
+        }
+        ${
+          args?.to_genesis_block_height
+            ? this.sql`AND gen.block_height <= ${args.to_genesis_block_height}`
+            : this.sql``
+        }
+        ${
+          args?.from_sat_coinbase_height
+            ? this.sql`AND loc.sat_coinbase_height >= ${args.from_sat_coinbase_height}`
+            : this.sql``
+        }
+        ${
+          args?.to_sat_coinbase_height
+            ? this.sql`AND loc.sat_coinbase_height <= ${args.to_sat_coinbase_height}`
+            : this.sql``
+        }
+        ${
+          args?.from_genesis_timestamp
+            ? this.sql`AND gen.timestamp >= to_timestamp(${args.from_genesis_timestamp})`
+            : this.sql``
+        }
+        ${
+          args?.to_genesis_timestamp
+            ? this.sql`AND gen.timestamp <= to_timestamp(${args.to_genesis_timestamp})`
+            : this.sql``
+        }
+        ${
+          args?.from_sat_ordinal
+            ? this.sql`AND loc.sat_ordinal >= ${args.from_sat_ordinal}`
+            : this.sql``
+        }
+        ${
+          args?.to_sat_ordinal
+            ? this.sql`AND loc.sat_ordinal <= ${args.to_sat_ordinal}`
+            : this.sql``
+        }
+        ${args?.number?.length ? this.sql`AND i.number IN ${this.sql(args.number)}` : this.sql``}
+        ${args?.from_number ? this.sql`AND i.number >= ${args.from_number}` : this.sql``}
+        ${args?.to_number ? this.sql`AND i.number <= ${args.to_number}` : this.sql``}
+        ${
+          args?.address?.length
+            ? this.sql`AND loc.address IN ${this.sql(args.address)}`
+            : this.sql``
+        }
+        ${
+          args?.mime_type?.length
+            ? this.sql`AND i.mime_type IN ${this.sql(args.mime_type)}`
+            : this.sql``
+        }
+        ${args?.output ? this.sql`AND loc.output = ${args.output}` : this.sql``}
+        ${
+          args?.sat_rarity?.length
+            ? this.sql`AND loc.sat_rarity IN ${this.sql(args.sat_rarity)}`
+            : this.sql``
+        }
+        ${args?.sat_ordinal ? this.sql`AND loc.sat_ordinal = ${args.sat_ordinal}` : this.sql``}
+      ORDER BY ${this.sql.unsafe(orderBy)} ${this.sql.unsafe(order)}
+      LIMIT ${page.limit}
+      OFFSET ${page.offset}
+    `;
```

Comment:
Unsafe should be unnecessary here. Alternative using sql fragment interpolation:

```suggestion
    // Sanitize ordering args because we'll use sql fragment interpolation to add them into the query.
    let orderBy = this.sql`gen.block_height`;
    switch (sort?.order_by) {
      case OrderBy.ordinal:
        orderBy = this.sql`loc.sat_ordinal`;
        break;
      case OrderBy.rarity:
        orderBy = this.sql`
          ARRAY_POSITION(ARRAY['common','uncommon','rare','epic','legendary','mythic'], loc.sat_rarity)
        `;
        break;
    }
    const order = sort?.order === Order.asc ? this.sql`ASC` : this.sql`DESC`;
    // Do we need a filtered `COUNT(*)`? If not, use the global inscription count from `chain_tip`.
    const unfiltered =
      args === undefined || Object.values(args).find(k => k !== undefined) === undefined;

    const results = await this.sql<({ total: number } & DbFullyLocatedInscriptionResult)[]>`
      SELECT
        i.genesis_id,
        i.number,
        i.mime_type,
        i.content_type,
        i.content_length,
        i.fee AS genesis_fee,
        gen.block_height AS genesis_block_height,
        gen.block_hash AS genesis_block_hash,
        gen.tx_id AS genesis_tx_id,
        gen.timestamp AS genesis_timestamp,
        gen.address AS genesis_address,
        loc.tx_id,
        loc.address,
        loc.output,
        loc.offset,
        loc.sat_ordinal,
        loc.sat_rarity,
        loc.timestamp,
        loc.value,
        loc.sat_coinbase_height,
        ${unfiltered ? '0 as total' : this.sql`COUNT(*) OVER() as total`}
      FROM inscriptions AS i
      INNER JOIN locations AS loc ON loc.inscription_id = i.id
      INNER JOIN locations AS gen ON gen.inscription_id = i.id
      WHERE loc.current = TRUE AND gen.genesis = TRUE
        ${
          args?.genesis_id?.length
            ? this.sql`AND i.genesis_id IN ${this.sql(args.genesis_id)}`
            : this.sql``
        }
        ${
          args?.genesis_block_height
            ? this.sql`AND gen.block_height = ${args.genesis_block_height}`
            : this.sql``
        }
        ${
          args?.genesis_block_hash
            ? this.sql`AND gen.block_hash = ${args.genesis_block_hash}`
            : this.sql``
        }
        ${
          args?.from_genesis_block_height
            ? this.sql`AND gen.block_height >= ${args.from_genesis_block_height}`
            : this.sql``
        }
        ${
          args?.to_genesis_block_height
            ? this.sql`AND gen.block_height <= ${args.to_genesis_block_height}`
            : this.sql``
        }
        ${
          args?.from_sat_coinbase_height
            ? this.sql`AND loc.sat_coinbase_height >= ${args.from_sat_coinbase_height}`
            : this.sql``
        }
        ${
          args?.to_sat_coinbase_height
            ? this.sql`AND loc.sat_coinbase_height <= ${args.to_sat_coinbase_height}`
            : this.sql``
        }
        ${
          args?.from_genesis_timestamp
            ? this.sql`AND gen.timestamp >= to_timestamp(${args.from_genesis_timestamp})`
            : this.sql``
        }
        ${
          args?.to_genesis_timestamp
            ? this.sql`AND gen.timestamp <= to_timestamp(${args.to_genesis_timestamp})`
            : this.sql``
        }
        ${
          args?.from_sat_ordinal
            ? this.sql`AND loc.sat_ordinal >= ${args.from_sat_ordinal}`
            : this.sql``
        }
        ${
          args?.to_sat_ordinal
            ? this.sql`AND loc.sat_ordinal <= ${args.to_sat_ordinal}`
            : this.sql``
        }
        ${args?.number?.length ? this.sql`AND i.number IN ${this.sql(args.number)}` : this.sql``}
        ${args?.from_number ? this.sql`AND i.number >= ${args.from_number}` : this.sql``}
        ${args?.to_number ? this.sql`AND i.number <= ${args.to_number}` : this.sql``}
        ${
          args?.address?.length
            ? this.sql`AND loc.address IN ${this.sql(args.address)}`
            : this.sql``
        }
        ${
          args?.mime_type?.length
            ? this.sql`AND i.mime_type IN ${this.sql(args.mime_type)}`
            : this.sql``
        }
        ${args?.output ? this.sql`AND loc.output = ${args.output}` : this.sql``}
        ${
          args?.sat_rarity?.length
            ? this.sql`AND loc.sat_rarity IN ${this.sql(args.sat_rarity)}`
            : this.sql``
        }
        ${args?.sat_ordinal ? this.sql`AND loc.sat_ordinal = ${args.sat_ordinal}` : this.sql``}
      ORDER BY ${orderBy} ${order}
      LIMIT ${page.limit}
      OFFSET ${page.offset}
    `;
```

Reply:
Lol changed a few lines in the `suggestion` block and it diffs the entire selection wtf

Reply:
Have you verified this? It seems to work as expected from the interpolation docs: https://github.com/porsager/postgres#quick-primer-on-interpolation

<img width="778" alt="image" src="https://user-images.githubusercontent.com/1447546/235747124-8b338792-d156-44b1-9dfe-3307dfd0ac31.png">

Reply:
This is very strange because according to the docs, that interpolation is a value interpolation (which produces a placeholder), I'm just trying your example and I get
```
syntax error at or near \"$3\"
```
I adjusted the first interpolation to be an identifier interpolation (which is correct) to `ORDER BY ${sql(orderBy)} ${order}` but then I get
```
syntax error at or near \"$2\"
```
Did you modify something on the `order` variable above?

Reply:
You want this code from the top of the suggestion:
```ts
    // Sanitize ordering args because we'll use sql fragment interpolation to add them into the query.
    let orderBy = this.sql`gen.block_height`;
    switch (sort?.order_by) {
      case OrderBy.ordinal:
        orderBy = this.sql`loc.sat_ordinal`;
        break;
      case OrderBy.rarity:
        orderBy = this.sql`
          ARRAY_POSITION(ARRAY['common','uncommon','rare','epic','legendary','mythic'], loc.sat_rarity)
        `;
        break;
    }
    const order = sort?.order === Order.asc ? this.sql`ASC` : this.sql`DESC`;
```

Reply:
Duh, I missed that, thanks 👍